.github/workflows/linters.yml

```diff
@@ -20,23 +20,25 @@ jobs:
         mkdir wp-content/themes/pub/wporg
         svn export https://meta.svn.wordpress.org/sites/trunk/wordpress.org/public_html/wp-content/themes/pub/wporg/css wp-content/themes/pub/wporg/css
 
+    - name: Get yarn cache directory path
+      id: yarn-cache-dir-path
+      run: echo "::set-output name=dir::$(yarn config get cacheFolder)"
+
     - name: Cache node modules
       uses: actions/cache@v2
-      env:
-        cache-name: cache-node-modules
+      id: yarn-cache
+
       with:
-        # npm cache files are stored in `~/.npm` on Linux/macOS
-        path: ~/.npm
-        key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+        path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+        key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+        
         restore-keys: |
-          ${{ runner.os }}-build-${{ env.cache-name }}-
-          ${{ runner.os }}-build-
-          ${{ runner.os }}-
+          ${{ runner.os }}-yarn-
 
-    - name: Use Node.js 12.x
+    - name: Use Node.js 14.x
       uses: actions/setup-node@v1
       with:
-        node-version: 12.x
+        node-version: 14.x
```

Comment:
Later versions of the `setup-node` action include [an optional yarn cache](https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#caching-packages-data), and can read from [a local `.nvmrc` file](https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#node-version-file) — for example, in [the wporg-main-2022 actions](https://github.com/WordPress/wporg-main-2022/blob/455a91e3165744864fe5b17933324a62b0be8d0d/.github/workflows/linters.yml#L19-L23). I like the `.nvmrc` file — it documents the node version and if/when we update, it only needs to be changed once.

Reply:
Thanks! I've updated the actions.
Curious why we're using a certain commit of setup-node instead of something like `@v3` in [the wporg-main-2022 actions](https://github.com/WordPress/wporg-main-2022/blob/455a91e3165744864fe5b17933324a62b0be8d0d/.github/workflows/linters.yml#L19-L23)? Maybe it didn't support it back then?

Reply:
I pulled it from the [wordpress-develop](https://github.com/WordPress/wordpress-develop/) repo. I believe they review the actions for security, and using a specific SHA ensures the version used is the one that was reviewed.

Reply:
Just found that they set up a dependabot to update the action/setup-node every time it has a new release.
https://github.com/WordPress/wordpress-develop/pull/3350/files

I'm thinking that it would be safe and up-to-date as well by following its [instruction](https://github.com/actions/setup-node#usage) to use `@v3` as well.